lib/IRGen/GenType.cpp

```diff
@@ -2289,8 +2289,17 @@ SILType irgen::getSingletonAggregateFieldType(IRGenModule &IGM, SILType t,
     auto allFields = structDecl->getStoredProperties();
     
     auto field = allFields.begin();
-    if (!allFields.empty() && std::next(field) == allFields.end())
-      return t.getFieldType(*field, IGM.getSILModule());
+    if (!allFields.empty() && std::next(field) == allFields.end()) {
+      auto fieldTy = t.getFieldType(*field, IGM.getSILModule());
+      if (auto fieldDecl = fieldTy.getNominalOrBoundGenericNominal()) {
```

Comment:
What if the field type is a tuple containing an inaccessible type? I think SIL Type lowering has a notion of an “ABI visible” type that handles such cases, and we just need to use the right API here

Reply:
Tuples, tuples, tuples!

https://github.com/apple/swift/pull/25259